sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaReadSchemaTransformProvider.java

```diff
@@ -262,17 +270,22 @@ public static class ErrorFn extends DoFn<byte[], Row> {
     private SerializableFunction<byte[], Row> valueMapper;
     private Counter errorCounter;
     private Long errorsInBundle = 0L;
+    private final boolean failOnError;
 
-    public ErrorFn(String name, SerializableFunction<byte[], Row> valueMapper) {
+    public ErrorFn(String name, SerializableFunction<byte[], Row> valueMapper, boolean failOnError) {
       this.errorCounter = Metrics.counter(KafkaReadSchemaTransformProvider.class, name);
       this.valueMapper = valueMapper;
+      this.failOnError = failOnError;
     }
 
     @ProcessElement
     public void process(@DoFn.Element byte[] msg, MultiOutputReceiver receiver) {
       try {
         receiver.get(OUTPUT_TAG).output(valueMapper.apply(msg));
       } catch (Exception e) {
+        if (failOnError) {
+          throw new RuntimeException(e.getMessage());
```

Comment:
This will lose the original context. You can throw new RuntimeException(e). Perhaps better would be to catch RuntimeException in a separate clause and throw the original.

Reply:
Done.